doc/adr/0010-atomic-composite-grapth-operations.md

```diff
@@ -0,0 +1,267 @@
+but are not really reusing these Atomic Ops.
+
+Implementing `:block/save` that sometimes is just updating `:block/string` and other times also needs to `:page/create`.
+`:paste` event is another that will be super hard to implement w/o Atomic Graph Operations.
+
+
+## Decision
+
+We have two kinds of events to modify graph:
+- ⚛️ Atomic Graph Ops
+    - Not divisible Graph Ops
+    - Operations like create new block, create page, save block
+- ⎄ Composite Graph Ops
+    - Collection of events to be executed on the graph
+    - Like `:block/save` when new link is discovered, should produce also `:page/create` event
+
+Atomic events should follow same Common Events model as it's happening now.
+
+List of ((⚛️ Atomic Graph Ops))
+- *Page Ops*
+    - ((⚛️**`:page/new`**))
+    - ((⚛️**`:page/rename`**))
+    - ((⚛️**`:page/merge`**))
+    - ((⚛️**`:page/remove`**))
+- *Block Ops*
+    - ((⚛️**`:block/new`**))
+    - ((⚛️**`:block/save`**))
+    - ((⚛️**`:block/open`**))
+    - ((⚛️**`:block/move`** ⭐️))
+    - ⚛️**`:block/remove`**
+- *Shortcut Ops*
+    - (({{[[TODO]]}} ⚛️**`:shortcut/new`**))
+    - (({{[[TODO]]}} ⚛️**`:shortcut/remove`**))
+    - (({{[[TODO]]}} ⚛️**`:shortcut/move`**))
+
+
+((⎄ Composite Graph Ops)) are a way to group ((⚛️ Atomic Graph Ops)).
+
+We'll only need one type of Composite Graph Ops, that is Consequence Event.
+
+Consequence Event is 2 things:
+- Trigger Event (like `:paste`)
+- Consequence Events List
+    - Containing however many events to include into transaction resolution.
+    - These events can be more of Consequence Events, or ((⚛️ Atomic Graph Ops)).
+
+
+This will allow for easier Temporality,
+We'll be able to represent Main Action that moved Knowledge Graph forward.
+
+It will maintain information about `:paste` events that injected sub-tree with ((⚛️ Atomic Graph Ops)).
+
+## Consequences
+
+We'll have additional work of Porting Semantic Events to use Atomic Graph Ops and compositions.
+
+We have working `:paste` & `:block/save` events, they are broken not ported now.
+
+We'll have smaller amount of tests in order to provide correctness guarantees
+
+
+## Additional Resources
+
+### Catalog of operations
+
+- Types of Graph Operations
+    - ⚛️ Atomic Graph Ops
+        - Not divisible Graph Ops
+        - Operations like create new block, create page, save block
+    - ⎄ Composite Graph Ops
+        - Collection of events to be executed on the graph
+        - Like `:block/save` when new link is discovered, should produce also `:page/create` event
+
+- List of Atomic Graph Operations:
+    - Page Ops
+        - ⚛️**`:page/new`**
+            - ((⚛️ Atomic Graph Ops))
+            - *Input*
+                - `title` - Page title page to be created
+                - `page-uid` - `:block/uid` of page to be created
+                - `block-uid` - `:block/uid` of 1st block to be created in page to be created
+        - ⚛️**`:page/rename`**
```

Comment:
I think creating a new page and renaming a page are actually the same operation. Partly because they take the same args (since I don't think the first block is actually part of page/new), and partly because the only way to distinguish in their usage is by verifying whether or not the page exists before.

If we make them the same atomic op we have a couple of extra benefits that are very desirable in a concurrent and partial information environment:
- idempotency
- doesn't require knowing if the page exists or not already

Reply:
Page creation does create 1st block in that page as well.
I can't see how page/create and page/rename are the same thing.